Shared/Supporting Files/Extensions/CategoryView+Search.swift

```diff
@@ -15,13 +15,51 @@
 import SwiftUI
 
 extension CategoryView {
-    /// Searches the sample list using the query.
-    /// - Returns: A set of samples matching the query criteria.
-    func searchSamples() -> [Sample] {
-        if query.isEmpty {
-            return samples
-        } else {
-            return samples.filter { $0.name.localizedCaseInsensitiveContains(query) }
+    /// Searches the samples using the sample's name and the query.
+    /// - Returns: The samples whose name partially matches the query.
+    func searchSamplesNames() -> [Sample] {
+        // Return all the samples when query is empty.
+        guard !query.isEmpty else { return samples }
+        
+        // Preform a partial text search using the sample's name and the query.
+        let nameSearchResults = samples.filter { sample in
+            sample.name.localizedCaseInsensitiveContains(query)
         }
+        return nameSearchResults
+    }
+    
+    /// Searches the samples using the sample's description and the query.
+    /// - Returns: The samples whose description partially matches the query.
+    func searchSamplesDescriptions() -> [Sample] {
+        // The samples already found by name with query.
+        let previousSearchResults = searchSamplesNames()
```

Comment:
This makes me think if you can use a state property `Set` to hold the temporary results (or the remainder of the samples to be searched), instead of cascading the results from previously run methods. Otherwise it would result in 1-2-3-4-… n-squared rounds of looping through the `samples` array as we add n criteria. 

> Should sections include results that were found previous section?

I'm not object to including duplicates in each section. Although in iOS, it is often represented by a segmented control-like UI. See Music app as an example.

Reply:
I added state property, but I am not sure if it was in the pattern you were thinking off. So, let me know what you think.